disk: defaultCollector missing collectInstantaneous breaks win-amd64 build

The `statsCollector` interface in `pkg/storage/disk/monitor.go` requires two methods: `collect` and `collectInstantaneous`. The `collectInstantaneous` method was added to the interface and implemented on `linuxStatsCollector` and `darwinCollector`, but was not added to `defaultCollector` in `platform_default.go` (used for `!linux && !darwin` builds, including Windows).

This causes the win-amd64 release build to fail:

```
pkg/storage/disk/monitor.go:101:26: cannot use collector (variable of type *defaultCollector) as statsCollector value in assignment: *defaultCollector does not implement statsCollector (missing method collectInstantaneous)
pkg/storage/disk/monitor.go:102:27: cannot use collector (variable of type *defaultCollector) as statsCollector value in argument to m.monitorDisks: *defaultCollector does not implement statsCollector (missing method collectInstantaneous)
```

Affected file: `pkg/storage/disk/platform_default.go`

Jira issue: CRDB-60765

--- a/pkg/storage/disk/platform_default.go
+++ b/pkg/storage/disk/platform_default.go
@@ -20,6 +20,12 @@ func (defaultCollector) collect(disks []*monitoredDisk, time time.Time) (int, er
 	return len(disks), nil
 }
 
+func (defaultCollector) collectInstantaneous(
+	disks []*monitoredDisk, now time.Time, recorder func(traceEvent), buf []byte,
+) (countCollected int, _ []byte, err error) {
+	return len(disks), buf, nil
+}
+
 func newStatsCollector(fs vfs.FS) (*defaultCollector, error) {
 	return &defaultCollector{}, nil
 }

